### ISSUE TITLE
Loss of "unwatched" tags in shared shows

### ISSUE THREAD

**Issue by WayneD** [plexinc/plex-media-player#227]
There is a bug when a user views some shared content and drills down to an episode page: it frequently causes all the show's "unwatched" tags to vanish. The owner of the content being shared doesn't have this issue. I've only seen it if a show has multiple unwatched episodes. Also, this bug doesn't lose unwatched info on the server, it just messes up the display of the data so that it becomes impossible to find unwatched episodes in a show once an episode page in that show has been viewed.

For instance, from the home page if I press Right twice to get to a show in the On Deck list that has a bunch of unwatched episodes, press Enter to view the episode information, then the "Unwatched" orange tag shows up very briefly and vanishes. At that point, all the unwatched tags are gone for all episodes in that show, including the triangle tags on the home page (e.g. the on-deck item has no triangle, even though it wouldn't be there if it were not unwatched).

I've tried moving my custom keymap file out of the way, tried with and without OpenGL enabled, and even tried without fullscreen enabled (just to see if it might affect some subtle memory corruption bug), but there was no change.

I'm attaching a log file where I open PMP, press right, right, enter, Ctrl-Q (I've just edited out my IP):
[log-of-the-issue.zip](https://github.com/plexinc/plex-media-player/files/120409/log-of-the-issue.zip)

**Comment by WayneD** [comment]
I dug into the server logs, turning on token logging so that I could try out the URLs being fetched and I think I figured out the issue. The server queries for "allLeaves" are returning the unwatched status of the master account instead of the shared account.  The queries being made are like this one:

`/library/metadata/11840/allLeaves?X-Plex-No-Cache=1455474258903625&X-Plex-Version=2.5.3&X-Plex-Provides=player&X-Plex-Product=Plex%20Media%20Player&X-Plex-Platform=Konvergo&X-Plex-Platform-Version=1.0.4.164-cc08ff9d&X-Plex-Device=Windows&X-Plex-Device-Name=argo&X-Plex-Model=10&X-Plex-Vendor=1280x1539%2C%202736x1824&X-Plex-Client-Identifier=6kx0shtcnvr6bt9&X-Plex-Token=xxxxxxxxxxxxxxxxxxxx`

If I twiddle the watched/unwatched status of various episodes in the master account and then drill down to the episode page in PMP in the shared account, all the unwatched flags change to match the master account.

**Comment by WayneD** [comment]
I've reported this bug to the PMS folks who will hopefully have the underlying issue with allLeaves fixed soon.  Since the fix isn't in PMP, I'll go ahead and close this.

As an aside: I first reported the weird behavior of PMP in the forums on November 30th, and was met with silence. I then post it here (hoping that this was a better place to report a bug) and was met with silence.I then tracked down the underlying cause of the issue and still there was silence (though one non-employee did respond in the forums at that point). So, there is room for improvement in bug responses, especially for something that ends up being a cross-team issue.

Fortunately my experience posting about the bug into the PMS forum was totally different -- I got a very quick response and it sounds like the issue has been identified and will be fixed soon.
